cms/djangoapps/contentstore/rest_api/v1/views.py

```diff
@@ -109,7 +110,18 @@ def get(self, request, course_id):
     def post(self, request, course_id):
         """ POST handler """
         exam_config = ProctoredExamSettingsSerializer(data=request.data.get('proctored_exam_settings', {}))
-        exam_config.is_valid(raise_exception=True)
+        valid_request = exam_config.is_valid()
```

Comment:
I wonder if you could simplify this a bit by determining which serializer to use earlier on. Then, the serializer could handle the validation for you. Here's roughly what I'm thinking.

```
 is_edX_staff = request.user.is_staff
serializer = ProctoredExamSettingsSerializer if is_edx_staff else RestrictedProctoredExamSettingsSerializer
exam_config = serializer(data=request.data.get('proctored_exam_settings', {}))
valid_request = exam_config.is_valid()

``` 

I think that the last line should raise a `400` if, for example, the user is not edX staff and tries to include restricted data in the post body. But perhaps `403` is a more semantically correct status code? What do you think?

Reply:
I had thought of doing something similar, but it wasn't raising a 400 if the user is not edX staff and includes restricted data. That's why I was breaking down cases by user permissions and if the initial request was valid.

Reply:
I still think @MichaelRoytman's comment is valid. But in order to detect the 400 case, you need to do this:
```
is_edx_staff = request.user.is_staff
serializer = ProctoredExamSettingsSerializer if is_edx_staff else RestrictedProctoredExamSettingsSerializer
exam_config = serializer(data=request.data.get('proctored_exam_settings', {}))
valid_request = exam_config.is_valid()
if not is_edx_staff and valid_request and ProctoredExamSettingsSerializer(data=request.data.get('proctored_exam_settings', {}).is_valid():
    return Response(status=status.HTTP_400_BAD_REQUEST)
```